frontend/src/components/Input/index.tsx

```diff
@@ -1,20 +1,85 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+
+/**
+ * Reusable input component with predefined style.
+ *
+ */
 
 interface InputProps {
     type: React.HTMLInputTypeAttribute;
     value: string | number | readonly string[] | undefined;
     onInput: React.FormEventHandler<HTMLInputElement>;
+    label?: string;
+    textHelper?: string;
+    required?: true;
 }
 
 // Input template with predefined styles
-export const CustomInput: React.FC<InputProps> = ({ type, value, onInput }) => {
+export const CustomInput: React.FC<InputProps> = ({
+    type,
+    value,
+    onInput,
+    label,
+    textHelper,
+    required,
+}) => {
+    const [touched, setTouched] = useState(false);
+    let error = '';
+
+    //Simple validation
+    if (touched && required && value === '') {
+        error = '*Required';
+    }
+
     return (
-        <input
-            type={type}
-            spellCheck={'false'}
-            value={value}
-            className=" font-sans text-base font-normal leading-normal p-3 rounded-xl text-gray-900 bg-white border-gray-300 border shadow hover:border-primary focus:outline-none focus:ring focus:border-primary"
-            onInput={onInput}
-        />
+        <div className="relative">
```

Comment:
Is there some element which is absolute? I don't think this needs to be `relative`.

Reply:
The label uses absolute positioning to position itself to the top corner of the input element